homeworks/asztaloslajos/5_oop/car_fleet_mgr.py

```diff
@@ -0,0 +1,104 @@
+"""
+to_do.py --- Asztalos Lajos --- 2024.11.01
+"""
+# Car
+class Car:
+    car_id = int(0)
+    def __init__(self, brand, model, year:int):
+        Car.car_id += 1
+        self.brand = brand
+        self.model = model
+        self.year = year
+        self.mileage = int(0)
+        self.fuel_level = float(100)
+        self._id = Car.car_id
+
+# Car str method update
+    def __str__(self):
+        return(f"Azonosító: {self._id}, Gyártó: {self.brand}, Modell: {self.model}, Gyártási év: {self.year}, Futott kilométer:{self.mileage} Üzemanyagszint: {self.fuel_level}%")
+
+# Car drive method
+    def drive(self, distance:int) -> bool:
+        self._fuel_consumption = distance * 0.1
+        if distance <= 0:
+            print("Nem megfelelő kilométerértéket adtál meg!\n")
+            return(False)
+        elif self.fuel_level-self._fuel_consumption >= 0:
+            self.fuel_level -= self._fuel_consumption
+            self.mileage += distance
+            return(True)
+        else:
+            print("Túl sok megtett kilométert adtál meg!\n")
+            return(False)
+
+# Car refuel method
+    def refuel(self,fuel:float) -> bool:
+        if fuel <=0:
+            print("Nem megfelelő üzemanyagtöltési értéket adtál meg!\n")
+            return(False)
+        elif self.fuel_level + fuel <=100:
+            self.fuel_level += fuel
+            return(True)
+        else:
+            print("Túl sok üzemanyagmennyiséget adtál meg!\n")
+            return(False)
+# Fleet
+class Fleet:
+
+    def __init__(self):
+        self.cars = []
+
+# Fleet list_car method  
+    def list_car(self) -> None:
+        print("Az autók listája:")
+        for car in self.cars:
+            print(car)
+
+# Fleet add_car method
+    def add_car(self, car: Car) -> bool:
+        self.cars.append(car)
+        return True
+
+# Fleet remove_car method   
+    def remove_car(self, car_id:int) -> bool:
```

Comment:
Nem hiba, csak egyszerűsítési lehetőség:

```
    def remove_car(self, car:Car) -> bool:
        if car in self.cars:
            self.cars.remove(car)
            print(f"{car} sikeresen eltávolítva a flottából")
            return True
        print("Nincs ilyen azonosítójú autó!\n")
        return False
```

Ha magát az objektumot adod meg mint törlendő autó, akkor simán megvizsgálhatod hogy az az objektum benne van e a listában (itt a háttérben az __eq__() method fut le amiről tanultunk, és ez alapértelmezetten is működik mert konkrétan ugyanarról az objektumról van szó, szóval stimmel a memory address.)